agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -83,12 +91,14 @@ type TaskEngineState interface {
 type DockerTaskEngineState struct {
 	lock sync.RWMutex
 
-	tasks          map[string]*api.Task                       // taskarn -> api.Task
-	idToTask       map[string]string                          // DockerId -> taskarn
-	taskToID       map[string]map[string]*api.DockerContainer // taskarn -> (containername -> api.DockerContainer)
-	idToContainer  map[string]*api.DockerContainer            // DockerId -> api.DockerContainer
-	eniAttachments map[string]*api.ENIAttachment              // ENIMac -> api.ENIAttachment
-	imageStates    map[string]*image.ImageState
+	tasks               map[string]*api.Task                       // taskarn -> api.Task
+	idToTask            map[string]string                          // DockerId -> taskarn
+	taskToID            map[string]map[string]*api.DockerContainer // taskarn -> (containername -> api.DockerContainer)
+	idToContainer       map[string]*api.DockerContainer            // DockerId -> api.DockerContainer
+	eniAttachments      map[string]*api.ENIAttachment              // ENIMac -> api.ENIAttachment
+	imageStates         map[string]*image.ImageState
+	containerIDToLabels map[string]map[string]string // container id -> label(key-> value)
```

Comment:
Why not make label part of the container struct?

Reply:
so, don't even have it state, but save it in `api.Container` similar to how I'm handling timestamps?